docs/zksync-era-integration.md

```diff
@@ -0,0 +1,184 @@
+
+These functions are specially used in the `era_vm` to calculate refunds and pubdata costs. See [here](https://github.com/lambdaclass/era_vm/blob/main/src/state.rs#L108-L123) and [here](https://github.com/lambdaclass/era_vm/blob/main/src/state.rs#L132-L173).
+
+Finally, here is the [full](https://github.com/lambdaclass/zksync-era/blob/era_vm_integration_v2/core/lib/multivm/src/versions/era_vm/vm.rs#L79-L154) vm initialization code.
+
+### Orchestrating the Bootloader
+
+The operator is responsible for managing the Bootloader within the `era_vm`. This includes injecting the bootloader code into the `era_vm` and maintaining its state by:
+
+-   Pushing transactions into the Bootloader.
+-   Passing necessary parameters into the Bootloader's memory.
+-   Starting Bootloader execution.
+-   Rolling back the VM state in case of errors.
+
+Since both transactions and the Bootloader run in the same `era_vm`, the bootloader accesses a reserved heap where the operator writes any required data. This interaction is continuous, as the bootloader is unaware of the broader state of the `era_vm`. To facilitate communication, the bootloader can write to a special address that triggers a suspension of the `era_vm` execution, allowing the operator to provide necessary data. These are known as **hooks**, and based on the written value a specific hook will get triggered by the operator. Here are some of the most important hooks:
+
+-   **PostResult**: Sets the last transaction result
+-   **TxHasEnded**: if the mode of execution is set to **OneTx**, then the execution is stopped and it returns the result collected in the _PostResult_ hook.
+-   **NotifyAboutRefunds**: Inform the operator about the amount of gas refunded after a transaction.
+-   **AskOperatorForRefund**: here the Bootloader asks the operator to suggest a refund amount for the transaction.
+-   **PubdataRequested**: At the end of the batch, the Bootloader asks for the data to publish on the l1 (more on this later).
+
+Now, where does the operator know where to write to? Again, within the `era_vm`, there exists a special heap reserved exclusively for the Bootloader. The Operator writes all the data in that heap which has designated slots based on the type of data to write (see more [here](https://github.com/lambdaclass/zksync-era/blob/era_vm_integration_v2/docs/specs/zk_evm/bootloader.md#structure-of-the-bootloaders-memory)). Transactions, for example, are pushed into the `[252189..523261]` slots.
+
+### Rollbacks and snapshots
+
+In the `era_vm`, when a transaction encounters a panic or reverts, the vm needs to roll back the changes, restoring only a part of the [state](https://github.com/lambdaclass/era_vm/blob/main/src/state.rs#L43-L60) to its previous frame. Remember that frames are created under `near_call` and `far_call` opcodes. Currently, rollbacks are perform using snapshots which are just copies of the current state. If a rollback is necessary, the state is restored from these snapshots.
```

Comment:
This section may be better suited [here](https://github.com/lambdaclass/era_vm/blob/main/docs/general.md#difference-between-a-revert-and-a-panic-exception-handlers).

Reply:
The idea was to recall the idea of rollbacks in `era_vm` to emphasize on the difference  with bootloader rollbacks.